compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2283,7 +2341,10 @@ public void visit(BLangRecordLiteral recordLiteral) {
 
                     if (names.contains(name)) {
                         this.dlog.error(keyExpr.pos, DiagnosticCode.DUPLICATE_KEY_IN_RECORD_LITERAL,
-                                        recordLiteral.expectedType.getKind().typeName(), name);
+                                recordLiteral.expectedType.getKind().typeName(), name);
+                    } else if (inclusiveTypeSpreadField != null && !neverTypedKeys.contains(name)) {
+                        this.dlog.error(keyExpr.pos, DiagnosticCode.POSSIBLE_DUPLICATE_KEY_VIA_INCLUSIVE_TYPE_SPREAD_OP,
```

Comment:
For me, this DiagnosticCode (and key) imply that the duplicate is via the spread field, while the actual error is that this field can duplicate a field specified via the spread field.

Shall we change these to something like `POSSIBLE_DUPLICATE_OF_FIELD_SPECIFIED_VIA_SPREAD_OP`?